recipes/crypto-rsa-pss-update/src/workflow.ts

```diff
@@ -0,0 +1,310 @@
+			if (valueText && /^['"]rsa-pss['"]$/.test(valueText)) {
+				return true;
+			}
+		}
+	}
+
+	// Also check let declarations
+	const letDeclarations = rootNode.findAll({
+		rule: {
+			pattern: `let ${identifier} = $VALUE`
+		}
+	});
+
+	for (const decl of letDeclarations) {
+		const valueMatch = decl.getMatch("VALUE");
+		if (valueMatch) {
+			const valueText = valueMatch.text()?.trim();
+			if (valueText && /^['"]rsa-pss['"]$/.test(valueText)) {
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Processes RSA-PSS generateKeyPair calls and transforms deprecated options
+ */
+function processRsaPssCalls(rootNode: SgNode<JS>, allCalls: SgNode<JS>[]): { edits: Edit[], hasChanges: boolean } {
+	const edits: Edit[] = [];
+	let hasChanges = false;
+
+	for (const call of allCalls) {
+		const typeMatch = call.getMatch("TYPE");
+		const optionsMatch = call.getMatch("OPTIONS");
+
+		if (!typeMatch || !optionsMatch) continue;
+
+		// Only process 'rsa-pss' key type
+		const typeText = typeMatch.text()?.trim();
+		if (!typeText) continue;
+
+		// Check if it's a string literal 'rsa-pss' or "rsa-pss"
+		const isStringLiteral = /^['"]rsa-pss['"]$/.test(typeText);
+		
+		// Check if it's a variable that resolves to 'rsa-pss'
+		const isVariableWithRsaPss = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/.test(typeText) && 
+			resolveVariableValue(rootNode, typeText);
+
+		if (!isStringLiteral && !isVariableWithRsaPss) {
+			continue;
+		}
+
+		// Find the object node that contains the options
+		const objectNode = optionsMatch.find({
+			rule: {
+				kind: "object"
+			}
+		});
+
+		if (!objectNode) continue;
+
+		// Find all property pairs in the object
+		const pairs = objectNode.findAll({
+			rule: {
+				kind: "pair"
+			}
+		});
+
+		// Transform hash and mgf1Hash properties
+		for (const pair of pairs) {
+				const keyNode = pair.find({
+					rule: {
+						any: [
+							{
+								regex: "hash",
+								kind: "property_identifier",
+							},
+							{
+								regex: "mgf1Hash",
+								kind: "property_identifier",
+							},
+						],
+					},
+				});
+				if (!keyNode) continue;
+				const key = keyNode.text()?.trim();
+				if (!key || (key !== "hash" && key !== "mgf1Hash")) continue;
+
+				// Get the value node to preserve it in the transformation
+				const valueNode = pair.find({
+					rule: {
+						kind: "string"
+					}
+				}) || pair.find({
+					rule: {
+						kind: "identifier"
+					}
+				}) || pair.find({
+					rule: {
+						kind: "template_string"
+					}
+				}) || pair.find({
+					rule: {
+						kind: "member_expression"  // For cases like crypto.constants.SHA256
+					}
+				}) || pair.find({
+					rule: {
+						kind: "call_expression"    // For cases like getHash()
+					}
+				}) || pair.find({
+					rule: {
+						kind: "binary_expression"  // For cases like 'sha' + '256'
+					}
+				}) || pair.find({
+					rule: {
+						kind: "ternary_expression"  // For cases like condition ? 'sha512' : 'sha256'
+					}
+				});
```

Comment:
Can we use any rule https://ast-grep.github.io/guide/rule-config/composite-rule.html#any